src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +111,29 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        clearInterval(this._pollingInterval);
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private setPollingMechanism(gatewayId: string | undefined, submitForm: any) {
```

Comment:
instead of `string | undefined` it's preferred to use `gatewayId?: string`
Also, can we have a type for `submitForm`? Using `any` is a bit like cheating.

Reply:
fixed